state/testing/suite.go

```diff
@@ -142,6 +145,19 @@ func (s *StateSuite) hubWatcherIdleFunc(modelUUID string) {
 	}
 }
 
+func (s *StateSuite) hubWatcherStartingFunc() {
+	s.hubWatcherStarting <- struct{}{}
```

Comment:
Do we want to do this or do we want:

s.hubWatcherStarted as the channel and then to do
```
close(s.hubWatcherStarted)
```
the main reason for close is if 0 or more units actually listen for the event, they can all get told, and it won't block the other goroutine. If you intentionally want the other routine to wait until we observe that it has started then what you have done is correct, except if there are 0 units that will ever listen for the event, then you have just caused this code to block indefinitely.
So you really need to wrap it in some sort of select{} that will see some other sort of event like tomb.Dying() or just a simple timeout so the test doesn't hang forever if you have a logic error.